.github/workflows/e2e-nvidia-l40s-x8.yml

```diff
@@ -39,14 +41,17 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: ${{ vars.AWS_REGION }}
 
-      - name: Start EC2 runner
-        id: start-ec2-runner
+      # Attempt to launch the EC2 runner on us-east-2b. If it fails to launch due to insufficient capacity
+      # for this instance, we will continue and try another availability zone within this region.
+      - name: Attempt to start EC2 runner on us-east-2b
+        id: start-ec2-runner-us-east-2b
         uses: machulav/ec2-github-runner@fcfb31a5760dad1314a64a0e172b78ec6fc8a17e # v2.3.6
+        continue-on-error: true
         with:
           mode: start
           github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
           ec2-image-id: ${{ vars.AWS_EC2_AMI }}
-          ec2-instance-type: g6e.48xlarge
+          ec2-instance-type: ${{ env.INSTANCE_TYPE }}
           subnet-id: subnet-024298cefa3bedd61
```

Comment:
@courtneypacheco would we want to handle these subnet-id and security-group-id values without exposing them to public, as a general security best practice ? it could be a env variable or a secret if we want to keep it more tightened, WDYT?

Reply:
A good idea but I think outside of the scope of what this PR is trying to achieve

Reply:
In fact there is already an issue for this: https://github.com/instructlab/instructlab/issues/2861

Reply:
fair point @nathan-weinberg. Thanks for pointing to the original ticket.

Reply:
I'd prefer to assign all of our AWS "configs" (subnet IDs, SG IDs, instance types, etc.) to variables, except in the few cases where it might not make sense.